vm/src/stdlib/syslog.rs

```diff
@@ -0,0 +1,181 @@
+    use num_traits::ToPrimitive;
+
+    #[pyattr]
+    const LOG_ALERT: i32 = libc::LOG_ALERT;
+    #[pyattr]
+    const LOG_AUTH: i32 = libc::LOG_AUTH;
+    #[cfg(not(target_os = "redox"))]
+    #[pyattr]
+    const LOG_AUTHPRIV: i32 = libc::LOG_AUTHPRIV;
+    #[pyattr]
+    const LOG_CONS: i32 = libc::LOG_CONS;
+    #[pyattr]
+    const LOG_CRIT: i32 = libc::LOG_CRIT;
+    #[cfg(not(target_os = "redox"))]
+    #[pyattr]
+    const LOG_CRON: i32 = libc::LOG_CRON;
+    #[pyattr]
+    const LOG_DAEMON: i32 = libc::LOG_DAEMON;
+    #[pyattr]
+    const LOG_DEBUG: i32 = libc::LOG_DEBUG;
+    #[pyattr]
+    const LOG_EMERG: i32 = libc::LOG_EMERG;
+    #[pyattr]
+    const LOG_ERR: i32 = libc::LOG_ERR;
+    #[pyattr]
+    const LOG_INFO: i32 = libc::LOG_INFO;
+    #[pyattr]
+    const LOG_KERN: i32 = libc::LOG_KERN;
+    #[pyattr]
+    const LOG_LOCAL0: i32 = libc::LOG_LOCAL0;
+    #[pyattr]
+    const LOG_LOCAL1: i32 = libc::LOG_LOCAL1;
+    #[pyattr]
+    const LOG_LOCAL2: i32 = libc::LOG_LOCAL2;
+    #[pyattr]
+    const LOG_LOCAL3: i32 = libc::LOG_LOCAL3;
+    #[pyattr]
+    const LOG_LOCAL4: i32 = libc::LOG_LOCAL4;
+    #[pyattr]
+    const LOG_LOCAL5: i32 = libc::LOG_LOCAL5;
+    #[pyattr]
+    const LOG_LOCAL6: i32 = libc::LOG_LOCAL6;
+    #[pyattr]
+    const LOG_LOCAL7: i32 = libc::LOG_LOCAL7;
+    #[pyattr]
+    const LOG_LPR: i32 = libc::LOG_LPR;
+    #[pyattr]
+    const LOG_MAIL: i32 = libc::LOG_MAIL;
+    #[pyattr]
+    const LOG_NDELAY: i32 = libc::LOG_NDELAY;
+    #[pyattr]
+    const LOG_NEWS: i32 = libc::LOG_NEWS;
+    #[pyattr]
+    const LOG_NOTICE: i32 = libc::LOG_NOTICE;
+    #[pyattr]
+    const LOG_NOWAIT: i32 = libc::LOG_NOWAIT;
+    #[pyattr]
+    const LOG_ODELAY: i32 = libc::LOG_ODELAY;
+    #[cfg(not(target_os = "redox"))]
+    #[pyattr]
+    const LOG_PERROR: i32 = libc::LOG_PERROR;
+    #[pyattr]
+    const LOG_PID: i32 = libc::LOG_PID;
+    #[pyattr]
+    const LOG_SYSLOG: i32 = libc::LOG_SYSLOG;
+    #[pyattr]
+    const LOG_USER: i32 = libc::LOG_USER;
+    #[pyattr]
+    const LOG_UUCP: i32 = libc::LOG_UUCP;
+    #[pyattr]
+    const LOG_WARNING: i32 = libc::LOG_WARNING;
+
+    fn get_argv(vm: &VirtualMachine) -> PyResult {
+        let argv = sysmodule::argv(vm);
+        let scriptobj = PyStrRef::try_from_object(vm, argv.get_item(0.into_pyobject(vm), vm)?)?;
+        let scriptobj = scriptobj.as_ref();
+
+        let scriptlen = scriptobj.len();
+        if scriptlen == 0 {
+            return Err(vm.new_value_error("empty script was given".to_owned()));
+        }
+
+        let retstr = match scriptobj.find('\\') {
+            Some(value) => scriptobj[value..scriptlen].to_string(),
+            None => scriptobj.to_string(),
+        };
+
+        Ok(retstr.into_pyobject(vm))
+    }
+
+    fn convert_cstring(s: &str, vm: &VirtualMachine) -> PyResult<std::ffi::CString> {
+        std::ffi::CString::new(s).map_err(|err| err.into_pyexception(vm))
+    }
```

Comment:
we already have `pystr::to_cstring` for `PyStr`, but nothing for `&str` itself. Creating a common utility for both usage in `vm/src/utils.rs` and sharing it would be great!

Reply:
@youknowone Thanks! I want to implement their shared trait as below code but something go wrong..
```rust
pub trait IntoCString<T: 'static + ?Sized> : AsRef<T> {
    fn to_cstring(&self, vm: &VirtualMachine) -> PyResult<std::ffi::CString> {
        std::ffi::CString::new(self.as_ref()).map_err(|err| err.into_pyexception(vm))
    }
}

impl IntoCString<str> for &str {}
impl IntoCString<str> for PyStr {}
```
Is there any idea for making this codes work?